02_bubble/hsy3418/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+/*
+ Basic bubble sort for an array of int
+*/
+func bubbleSort(s []int) []int {
+	for i := 1; i < len(s); i++ {
+		for j := 1; j < len(s); j++ {
```

Comment:
nit: 
```j<len(s)- i``` is more efficient
because the last elements are already sorted
also storing len(s) in a variable might be a good idea because it's calculating every time the loop loops

Reply:
I guess this comment doesn't really need addressing; it's still a bubble sort algorithm

Reply:
@Joshcarp  Thanks Joshua for such a detailed review, it's very very helpful!